src/Context/ProductsProvider.jsx

```diff
@@ -0,0 +1,93 @@
+import axios from "axios";
+import { createContext, useContext, useEffect, useReducer } from "react";
+
+const productContext = createContext([]);
+
+const productsReducer = (productState, action) => {
+  switch (action.type) {
+    case "getproducts":
+      return {
+        ...productState,
+        products: action.payload,
+      };
+    case "getcategories":
+      return {
+        ...productState,
+        categories: action.payload,
+      };
+    case "getbrands":
+      return {
+        ...productState,
+        brands: action.payload,
+      };
+  }
+};
+
+const ProductsProvider = ({ children }) => {
+  const [productState, productDispatch] = useReducer(productsReducer, {
+    products: JSON.parse(sessionStorage.getItem("productList")) ?? [],
+    categories: JSON.parse(sessionStorage.getItem("categories")) ?? [],
+    brands: JSON.parse(sessionStorage.getItem("brands")) ?? [],
+  });
+
+  useEffect(() => {
+    const getproducts = async () => {
+      try {
+        const response = await axios.get("/api/products");
+        productDispatch({
+          type: "getproducts",
+          payload: response.data.products,
+        });
+        sessionStorage.setItem(
+          "productList",
+          JSON.stringify(productState.products)
+        );
+      } catch (error) {
+        console.log(error);
```

Comment:
Suggestion : to elaborate on the error part use error.msg or error.response.data

Reply:
Noted.